### PR TITLE
ci: refactor downstream projects

### PR DESCRIPTION
**Pull request by yihau**
#### Summary of Changes

- separate downstream projects test into different jobs. I grouped them by project atm. (spl, example-helloworld, openbook-dex)

- move buildkite stuff into `.buildkite` folder

### CODE REVIEW

**Comment by t-nelson** [comment]
is this pretty much just reorganization, or are there functional changes in here?

**Comment by CriesofCarrots** [comment]
What's the motivation for this reorg?

**Comment by yihau** [comment]
@t-nelson It is more like a reorganization. I just moved those three functions into three different files. The only change I did was that I made [these ENVs](https://github.com/solana-labs/solana/pull/30463/files#diff-9e32bfa70d8ed6d9893c52f065356c3009ebe5207367af728d7e37df030384f6R9-R14) uppercase for fitting the shell naming convention.

**Comment by yihau** [comment]
@CriesofCarrots The main purpose for this PR is that I would like to run our downstream projects's tests simultaneously so that we can read/fix them easialy. it looks like this

Screenshot 2023-02-24 at 12 11 49 PM


I tried to separate those functions and keep original behavior. I'm going to create more PRs for ci scripts. My goal is that:

1. move all buildkite pipeline building stuff into `.buildkite`
2. make ci scripts can be run simultaneously and keeping the original behavior.

**Review by t-nelson — APPROVED** [review]
> @t-nelson It is more like a reorganization. I just moved those three functions into three different files. The only change I did was that I made [these ENVs](https://github.com/solana-labs/solana/pull/30463/files#diff-9e32bfa70d8ed6d9893c52f065356c3009ebe5207367af728d7e37df030384f6R9-R14) uppercase for fitting the shell naming convention.

thought so! thanks for clarifying

lgtm!

**Review by CriesofCarrots — APPROVED** [review]
Thanks for the context! Lgtm

File: ci/downstream-projects/func-spl.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    echo "used solana version: $project_used_solana_version"
+    if semverGT "$project_used_solana_version" "$SOLANA_VER"; then
+      echo "skip"
+      return
+    fi
+
+    ./patch.crates-io.sh "$SOLANA_DIR"
+
+    for program in "${PROGRAMS[@]}"; do
+      $CARGO_BUILD_SBF --manifest-path "$program"/Cargo.toml
```
Comment (joncinque) [inline ci/downstream-projects/func-spl.sh:37]: This used to be `$cargo_test_sbf` -- did it get changed to `build-sbf` for a reason?

Comment (yihau) [reply ci/downstream-projects/func-spl.sh:37]: thank you for pointing this out! I think it should be a bug 🐛 just open a PR to fixed it https://github.com/solana-labs/solana/pull/31363

**Pull request merged**